File: Android/shared/src/main/kotlin/shared/dto/objects/Device.kt
@file:UseSerializers(UUIDSerializer::class)

package shared.dto.objects

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.enums.SystemType
import shared.dto.enums.VersionType
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable
import shared.serializers.UUIDSerializer
import java.util.*

object Device {
    data object Parameters {
        @Serializable
        data class System(
            val manufacturer: String?,
            val model: String?,
            val system: SystemType?,
            val appVersion: String?,
            val country: String?,
            val fcmToken: String?,
        ) : Parametable()
    }

    data object Responses {
        @Serializable
        data class Full(
            val id: UUID,
            val version: VersionType?,
        ) : Responsable
    }
}

File: IOS/DTOs/Objects/Device.swift
import Foundation
import MemberwiseInit

public enum Device {
    public enum Parameters {}
    public enum Responses {}
}

public extension Device.Parameters {

    @MemberwiseInit(.public)
    struct System: Parametable {
        public var manufacturer : String?
        public var model: String?
        public var system: SystemType?
        public var appVersion: String?
        public var country: String?
        public let fcmToken: String?
    }
}

public extension Device.Responses {
    
    @MemberwiseInit(.public)
    struct Full: Responsable {
        public var id: UUID
        public var version: VersionType?
    }
}
